### PR TITLE
Added changes to introduce annotations during deploy

### PR DESCRIPTION
**Pull request by koushdey**

### CODE REVIEW

File: src/constants.ts [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
+
+export let pipelineAnnotations = [
+    `run=${process.env['GITHUB_RUN_ID']}`,
+    `workflow="${process.env['GITHUB_WORKFLOW']}"`,
+    `jobName="${process.env['GITHUB_JOB']}"`,
+    `createdBy=${process.env['GITHUB_ACTOR']}`,
+    `runUri=https://github.com/${process.env['GITHUB_REPOSITORY']}/actions/runs/${process.env['GITHUB_RUN_ID']}`,
+    `commit=${process.env['GITHUB_SHA']}`,
+    `branch=${process.env['GITHUB_REF']}`
+];
```
Comment (ds-ms) [inline src/constants.ts:36]: Is this an `LF --> CRLF` change? Why does it show like you've edited the entire file?

File: src/constants.ts [hunk 2]
```diff
@@ -1,26 +1,36 @@
...
+export const workloadTypesWithRolloutStatus: string[] = ['deployment', 'daemonset', 'statefulset'];
+
+export let pipelineAnnotations = [
+    `run=${process.env['GITHUB_RUN_ID']}`,
+    `workflow="${process.env['GITHUB_WORKFLOW']}"`,
+    `jobName="${process.env['GITHUB_JOB']}"`,
+    `createdBy=${process.env['GITHUB_ACTOR']}`,
+    `runUri=https://github.com/${process.env['GITHUB_REPOSITORY']}/actions/runs/${process.env['GITHUB_RUN_ID']}`,
+    `commit=${process.env['GITHUB_SHA']}`,
+    `branch=${process.env['GITHUB_REF']}`
```
Comment (ds-ms) [inline src/constants.ts:35]: Are these annotations final?

File: src/constants.ts (hunk 1 above)
Comment (koushdey) [reply src/constants.ts:36]: I was getting ^M in my edited lines. Hence converted from CRLF->LF. I tried to revert the format after the commit push. Still the same.

File: src/constants.ts (hunk 2 above)
Comment (koushdey) [reply src/constants.ts:35]: There is no ongoing discussion for the annotations. I will confirm and revert.

File: src/constants.ts (hunk 1 above)
Comment (ds-ms) [reply src/constants.ts:36]: Pointing out your specific changes is a bit tedious when it shows that you've edited the entire file. 😞

File: src/utilities/resource-object-utility.ts [hunk 3]
```diff
@@ -81,6 +81,34 @@ export function updateObjectLabels(inputObject: any, newLabels: Map<string, stri
     }
 }
 
+export function updateObjectAnnotations(inputObject: any, newAnnotations: Map<string, string>, override: boolean) {
```
Comment (koushdey) [inline src/utilities/resource-object-utility.ts:84]: Please review if these changes are necessary.

File: src/constants.ts [hunk 4]
```diff
@@ -24,3 +24,13 @@ export class ServiceTypes {
...
+
+export let workflowAnnotations = [
+    `run=${process.env['GITHUB_RUN_ID']}`,
+    `workflow="${process.env['GITHUB_WORKFLOW']}"`,
+    `jobName="${process.env['GITHUB_JOB']}"`,
+    `createdBy=${process.env['GITHUB_ACTOR']}`,
+    `runUri=https://github.com/${process.env['GITHUB_REPOSITORY']}/actions/runs/${process.env['GITHUB_RUN_ID']}`,
+    `commit=${process.env['GITHUB_SHA']}`,
+    `branch=${process.env['GITHUB_REF']}`
+];
```
Comment (ds-ms) [inline src/constants.ts:36]: These annotations are final?

File: src/utilities/utility.ts [hunk 5]
```diff
@@ -47,6 +50,29 @@ export function checkForErrors(execResults, warnIfError?: boolean) {
...
 }
 
+export function annotateChildPods(kubectl: Kubectl, resourceType: string, resourceName: string, allPods): IExecSyncResult[] {
+    const commandExecutionResults = [];
+    let owner = resourceName;
+    if (resourceType.toLowerCase().indexOf('deployment') > -1) {
+        owner = kubectl.getNewReplicaSet(resourceName);
+    }
+
+    if (!!allPods && !!allPods.items && allPods.items.length > 0) {
```
Comment (ds-ms) [inline src/utilities/utility.ts:60]: `!!` is redundant

**Comment by ds-ms** [comment]
Closing this PR in reference to #42

**Pull request closed without merging**